0000-miner-update.md

```diff
@@ -0,0 +1,199 @@
+choosing them?
+
+While there are several third party firmware update services like NervesHub,
+Mender and balena available none of them meet our immediate needs. NervesHub
+relies on Buildroot-based-images and Mender also has support for Buildroot but
+both of these services only allow for monolithic image updates like our existing
+nexthaul/nextgate mechanism. Balena deploys updates via Docker containers but it
+looks like the target edge devices need to run their Yocto-based balenaOS to
+receive them.
+
+- What is the impact of not doing this?
+
+If we do not allow Hotspot owners to build and distribute their own custom miner
+software we eventually risk greater scrutiny by Hotspot operators, regulators
+and large cryptocurrency exchanges.
+
+# Unresolved Questions
+[unresolved]: #unresolved-questions
+
+- What parts of the design do you expect to resolve through the HIP process
+before this gets merged?
+
+What is the minimum viable product we can implement without sacrificing security
+or diminishing the user experience?
+
+- What parts of the design do you expect to resolve through the implementation
+of this feature?
+
+With the exception of image overlays all of the approaches being considered are
+new so the only way to assess them is through rapid prototyping.
+
+- What related issues do you consider out of scope for this HIP that could be
+addressed in the future independently of the solution that comes out of this HIP?
+
+How does a third party package and sign a miner release for download and
+verification by a consumer Hotspot?
+
+How do we facilitate the deployment of miner instances in the cloud that receive
+LoRa packets from third party edge radio gateways?
+
+# Deployment Impact
+[deployment-impact]: #deployment-impact
+
+- How will current users be impacted?
+
+End users will have the option to specify the URL of a miner package repository
+or image registry besides Helium's so that subsequent miner updates originate
+from that third party miner package repository or image registry. If no
+alternative URL is specified then the miner package repository or image registry
+utilized by a Hotspot defaults to Helium's. In either case miner updates continue
+to happen automatically so an end users should notice no difference unless they
+select an unreliable miner package or image supplier.
+
+- How will existing documentation/knowledgebase need to be supported?
+
+The final solution will need to include clear step-by-step instructions for
+miner package or image maintainers. This documentation will likely take the form
+of a GitHub project README for a Helium minerhaul repo that miner package
+maintainers will need to fork to deliver their updates.
+
+UI for entering the URL of an alternative miner package repository will need to
+be added to the Hotspot Settings page of the Helium mobile app. A related help
+article will also need to be added to the support section of Helium's website
+for end users.
```

Comment:
From @cokes518 

> We've seen previously that if the miner is in a crash loop on the Hotspot, a user would not be able to pair with it over BLE and thus could never get to their Hotspot Settings to change or revert the URL of a non-helium repo.
> 
> There's a high likelihood of this happening. I'd suggest including the ability for a user perform a button-boot to revert to a working version of Helium miner.

Reply:
That's a good point. A user can still roll back to a previously working version of the Hotspot firmware by button-booting. However, a better fix is for not being able to pair over BLE is for the GATT server to read a Hotspot's name from the ECC rather than ask miner for it.

Reply:
From @Vagabond 

> Yeah, we should fix this irregardless so a working miner is not required for BLE to work.

Reply:
Working on it.  See [helium/gateway-config#54](https://github.com/helium/gateway-config/pull/54).